tests/testapp/tests/test_connection.py

```diff
@@ -0,0 +1,134 @@
+        with mock.patch.object(Connection, '_request', return_value=mock_object):
+            with mock.patch.object(Certificate, 'verify', return_value=True):
+                    return func(*args, **kwargs)
+    return wrapper
+
+
+class NetworkSyncConnectionTestCase(TestCase):
+
+    def setUp(self):
+        self.profile = "facilitydata"
+
+        self.root_scope_def = ScopeDefinition.objects.create(
+            id="rootcert",
+            profile=self.profile,
+            version=1,
+            primary_scope_param_key="mainpartition",
+            description="Root cert for ${mainpartition}.",
+            read_filter_template="",
+            write_filter_template="",
+            read_write_filter_template="${mainpartition}",
+        )
+
+        self.subset_scope_def = ScopeDefinition.objects.create(
+            id="subcert",
+            profile=self.profile,
+            version=1,
+            primary_scope_param_key="",
+            description="Subset cert under ${mainpartition} for ${subpartition}.",
+            read_filter_template="${mainpartition}",
+            write_filter_template="${mainpartition}:${subpartition}",
+            read_write_filter_template="",
+        )
+
+        self.root_cert = Certificate.generate_root_certificate(self.root_scope_def.id)
+
+        self.subset_cert = Certificate(
+            parent=self.root_cert,
+            profile=self.profile,
+            scope_definition=self.subset_scope_def,
+            scope_version=self.subset_scope_def.version,
+            scope_params=json.dumps({"mainpartition": self.root_cert.id, "subpartition": "abracadabra"}),
+            private_key=Key(),
+        )
+        self.root_cert.sign_certificate(self.subset_cert)
+        self.subset_cert.save()
+
+        self.controller = MorangoProfileController('facilitydata')
+        self.network_connection = self.controller.create_network_connection('127.0.0.1')
+
+    @mock_patch_decorator
+    def test_creating_sync_session_successful(self):
+        self.assertEqual(SyncSession.objects.filter(active=True).count(), 0)
+        self.network_connection.create_sync_session(self.subset_cert, self.root_cert)
+        self.assertEqual(SyncSession.objects.filter(active=True).count(), 1)
+
+    @mock_patch_decorator
+    def test_creating_sync_session_cert_fails_to_verify(self):
+        Certificate.verify.return_value = False
+        with self.assertRaises(CertificateSignatureInvalid):
+            self.network_connection.create_sync_session(self.subset_cert, self.root_cert)
+
+    @mock_patch_decorator
+    def test_get_remote_certs(self):
+        # mock certs being returned by server
+        certs = self.subset_cert.get_ancestors(include_self=True)
+        cert_serialized = json.dumps(CertificateSerializer(certs, many=True).data)
+        Connection._request.return_value = mock.Mock(data=cert_serialized)
+
+        # we want to see if the models are created (not saved) successfully
```

Comment:
Wait, don't they already exist, since you got these from the DB, and didn't delete the ones on disk before saving the new ones?

Reply:
I'm taking the "data" from the request and seeing if the cert models are inflated correctly (not saving them) and if they match the original certs "sent".